docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+# The autorenew feature
+
+## Goal
+-	Allow Hedera Services to automatically search for entities such as accounts, files, smart contracts, topics, tokens... that are about to expire and renew them.
+
+## Design
+- Introduce new settings in `application.properties`:
+  * `autorenew.isEnabled`
+  * `autorenew.numberOfEntitiesToCheck`
+  * `autorenew.maxNumberOfEntitiesToRenew`
+  * `autorenew.gracePeriod`
+- Each Hedera entity has an `expirationTime` which is the effective consensus timestamp at (and after) which the entity is set to expire.
+- Each Hedera entity also has an `autoRenewAccount` which is the account to pay for the fee at renewal. This `autoRenewAccount` could be itself if the entity is a crypto account or an account associated with the entity when it was created.
+- When a Hedera entity is created, its initial lifetime is defined by its `autoRenewPeriod`. At its `expirationTime`, Hedera Services will try to extend an entity's lifetime by another `autoRenewPeriod` if the `autoRenewAccount` has enough balance to do so, or as much extension as the remaining balance permits.
+- After handling a transaction, Hedera Services will search within the next `autorenew.numberOfEntitiesToCheck` for upto `autorenew.maxNumberOfEntitiesToRenew` entities that expired then try to renew these entities.
+- After the grace period, if the `expirationTime` of an entity is not extended, it will be deleted from the system.
+
+## Implementation
+Hedera Services will perform a circular scanning of entities, meaning after we reach the last entity in the system, we will go back scanning from the first entity in the system.
+
+When trying to renew an entity:
+1. Calculate the fee to extend the entity's `expirationTime` for another `autoRenewPeriod`.
+2. If the `autoRenewAccount` of the entity has enough balance to cover this fee:
+  - extend the entity's `expirationTime` for another `autoRenewPeriod`.
+  - otherwise, translate the remaining balance of the `autoRenewAccount` into an extension, preferably proportional to the fee calculated in step 1, then extend accordingly.
+3. If the grace period also passes, delete the entity from the system.
+
+For restart and reconnect: The last scanned entity must be in the state for synchronization.
+
+## Autorenewal record
+After autorenewing an entity, Hedera Services will generate a `TransactionRecord` that serves as an autorenewal record and contains the following:
+
+| Field | Type | Label | Description |
+|---|---|---|---|
+| receipt | TransactionReceipt | | receipt will contain either an accountID, a fileID, a contractID, a topicID or a tokenID that got autorenewed |
```

Comment:
Can schedules be renewed? 
If so there might be some implications given the default multi signatory agreement nature of the scheduled transaction. A schedule with an extended expiration time may have very different business implications than the original agreed upon/signed transaction.
As such for a `Schedule` entity you might want to have multiple account agreement before extending a `Schedule`. 
Unless we are explicit that the `autoRenewAccount` is intended for an impartial observer of the Schedule and not a signatory with incentive to extend/not extend.

Reply:
No schedule transactions don't autorenew.

Reply:
Okay thanks for the clarification. Makes sense to not renew based on the complexities above and more.